_layouts/docwithnav.html

```diff
@@ -164,49 +163,79 @@
+                              <span>{{ navbar.bigheader }}</span>
+                            </a>
+                        </h4>
+                      </div>
+                      {% assign availableToc = navbar.section | where_exp: "item", "item.title != navbar.bigheader" %}
+                      {% assign tocPrefix = navbar_id | append: "nav" %}
+                      <div id="{{ navbar_id }}" class="panel-collapse collapse {% if this_pages_navbar == navbar and availableToc.size > 0 %}in{% endif %}">
+                          {% if availableToc %}
+                            <div class="panel-body">
+                                {% include sidebar.html sidebar=availableToc prefix=tocPrefix %}
+                            </div> 
+                          {% endif %}
+                      </div>
+                    </div>
+                  {% endfor %}
+                </div>
+                <div class="row">
+                  <div class="col-sm-0"></div>
+                </div>
               </div>
-            </div>
-            <div id="sidebar">
-            	{% include sidebar.html sidebar=this_pages_navbar.toc prefix="nav" %}
             </div>
           </div>
-          <div class="col-xs-12 col-sm-8 col-md-9">
+          <div class="article-container">
             <div id="content-main">
+              <p class="visible-xs clearfix">
+                <button type="button" class="btn btn-primary btn-xs" data-toggle="offcanvas">Toggle nav</button>
+              </p>
               <!-- TOP DOCUMENTATION PAGE NAVBAR -->
-              <div style="text-align: left; padding: 18px 0 9px 0; min-width: 500px; overflow-x: hidden;">
-
-              <a data-proofer-ignore href="https://github.com/projectcalico/calico/tree/master/{{page.path}}" style="color: #999; text-decoration: none; float: right;" target="_blank"><i class="fa fa-github" aria-hidden="true"></i> Edit this page</a></div>
+              {% capture edit_page %}
+                  <a data-proofer-ignore href="https://github.com/projectcalico/calico/tree/master/{{page.path}}" class="edit-page-link" target="_blank">
+                    <i class="fa fa-github" aria-hidden="true"></i>
+                    Edit this page
+                  </a>
+              {% endcapture %}
               <!-- END TOP DOCUMENTATION PAGE NAVBAR -->
+              {% include nav-breadcrumbs.html content=edit_page %}
               {% include version_warning.html %}
               {% if page.show_title -%}
               <h1>{{ page.title }}</h1>
               {% endif -%}
+              {% if page.show_read_time -%}
+                {% include read_time.html %}
+              {% endif -%}
+              <div class="visible-xs visible-sm">
+                {% include toc.html html=content class=".toc" item_class=".toc__nested" %}
+              </div>
+              {% include section_index.html section_index=navbars %}
```

Comment:
I would have preferred we call this directly from any page that is trying to show it instead of calling it here and 'sometimes showing it'. That would enable us to be more flexible with which text we show before or after it.

I'm guessing this was done so we don't have to calculate `navbars` a second time inside those pages? But the `section_index.html` include uses a tiny subset of the `navbars` data to render the page.

Did you consider this at all? I don't feel strongly enough about this to need any changes but just wanted to get your input.